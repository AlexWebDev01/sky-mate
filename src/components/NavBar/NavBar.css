.nav-bar {
  display: flex;
  flex-direction: row;
  width: 100%;
  margin-top: 21px;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 0;
  left: 0;
}
.nav-bar .logo-link {
  z-index: 1;
}
.nav-bar .search-bar {
  z-index: 1;
}/*# sourceMappingURL=NavBar.css.map */