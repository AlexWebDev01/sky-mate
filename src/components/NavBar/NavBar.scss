.nav-bar {
    display: flex;
    flex-direction: row;
    width: 100%;
    margin-top: 21px;
    justify-content: center;
    align-items: center;
    position: relative;

    .logo-link {
        z-index: 1;
    }

    .search-bar {
        z-index: 1;
    }
}