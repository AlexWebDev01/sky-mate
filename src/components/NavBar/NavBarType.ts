export default interface NavBarType {
  handleSearch: (location: string) => void;
  pageStyle: string;
}
