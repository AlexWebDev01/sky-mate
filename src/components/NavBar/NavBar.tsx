
import SearchBar from "../SearchBar/SearchBar";
import NavBarType from "./NavBarType";
import Logo from "../Logo/Logo";

import './NavBar.css';

const NavBar: React.FC<NavBarType> = ({ handleSearch, pageStyle }) => {

    return(
        <div className="nav-bar">
            <Logo />
            <SearchBar handleSearch={handleSearch} pageStyle={pageStyle}/>
        </div>
    )
};

export default NavBar;