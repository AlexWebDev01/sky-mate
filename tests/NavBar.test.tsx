import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import NavBar from '@components/NavBar/NavBar';

jest.mock('@components/SearchBar/SearchBar', () => () => <div>SearchBar</div>);
jest.mock('@components/Logo/Logo', () => () => <div>Logo</div>);

describe('NavBar', () => {
  test('renders without crashing', () => {
    render(<NavBar />);
    expect(screen.getByTestId('nav-bar')).toBeInTheDocument();
  });
});
