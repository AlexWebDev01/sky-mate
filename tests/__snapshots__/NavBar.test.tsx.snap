// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="nav-bar"
    data-testid="nav-bar"
  >
    <div>
      Logo
    </div>
    <div>
      SearchBar
    </div>
  </div>
</DocumentFragment>
`;
